src/coredump/coredumpctl.c

```diff
@@ -938,9 +947,20 @@ static int run_debug(int argc, char **argv, void *userdata) {
                         arg_debugger = "gdb";
         }
 
-        debugger = strdup(arg_debugger);
-        if (!debugger)
-                return -ENOMEM;
+        r = strv_extend(&debugger_call, arg_debugger);
+        if (r < 0)
+                return log_oom();
+
+        if (arg_debugger_args) {
+                _cleanup_strv_free_ char **debugger_args = NULL;
+
+                r = strv_split_full(&debugger_args, arg_debugger_args, WHITESPACE, EXTRACT_UNQUOTE);
+                if (r < 0)
+                        return log_oom();
```

Comment:
usually we parse/validate arguments fully when parsing the cmdline. hence, please make arg_debugger_args an "strv", i.e. an array of strings (free it using STATIC_DESTRUCTOR_REGISTER(), see the prior art above), and call strv_split_full() already in parse_argv[]).